### ISSUE TITLE
ldconfig: Changing access rights of /run/ld-so-cache-dir/...~ to 0644 failed: No such file or directory

### ISSUE THREAD

**Issue by refi64** [flatpak/flatpak#3029]
## Linux distribution and version

Fedora Silverblue 30.

## Flatpak version

1.4.2.

## Description of the problem

Randomly showing up with my Chromium Flatpak, when a `flatpak-spawn --sandbox` instance is run:

```
ldconfig: Changing access rights of /run/ld-so-cache-dir/baa04b8f378d418299a60f7a29ab3c82b55d79085c57e4b0c6684252705fefaa~ to 0644 failed: No such file or directory
```

On one occasion this caused the new tab page to crash, though I can't say much else right now.

## Steps to reproduce

I'll get better steps to reproduce SOON(tm), but I figured that it might be worthwhile to file this now just in case someone recognizes the error.

**Comment by refi64** [comment]
So I was building Chrome again, but once I tried redirecting all my pkg-config calls into the sandbox, I kept seeing this happen over and over again. That's when I was able to figure out how to reliably reproduce this:

```
for x in {1..50}; do flatpak run org.freedesktop.Platform//19.08 -c 'echo 123' & done
```

This just runs 50 short-lived Flatpak processes in parallel. Upon running this, you can see a plethora of errors appear:

```
ldconfig: ldconfig: Renaming of /run/ld-so-cache-dir/112bb263f675d03860a13039cd49f7b27bc1750a1838021ad617121117e4ff2c~ to /run/ld-so-cache-dir/112bb263f675d03860a13039cd49f7b27bc1750a1838021ad617121117e4ff2c failedChanging access rights of /run/ld-so-cache-dir/112bb263f675d03860a13039cd49f7b27bc1750a1838021ad617121117e4ff2c~ to 0644 failed: No such file or directory
: No such file or directory
ldconfig: Renaming of /run/ld-so-cache-dir/112bb263f675d03860a13039cd49f7b27bc1750a1838021ad617121117e4ff2c~ to /run/ld-so-cache-dir/112bb263f675d03860a13039cd49f7b27bc1750a1838021ad617121117e4ff2c failed: No such file or directory
error: ldconfig failed, exit status 256
error: ldconfig failed, exit status 256
error: ldconfig failed, exit status 256
ldconfig: Changing access rights of /run/ld-so-cache-dir/112bb263f675d03860a13039cd49f7b27bc1750a1838021ad617121117e4ff2c~ to 0644 failed: No such file or directory
error: ldconfig failed, exit status 256
ldconfig: Renaming of /run/ld-so-cache-dir/112bb263f675d03860a13039cd49f7b27bc1750a1838021ad617121117e4ff2c~ to /run/ld-so-cache-dir/112bb263f675d03860a13039cd49f7b27bc1750a1838021ad617121117e4ff2c failed: No such file or directory
error: ldconfig failed, exit status 256
ldconfig: Renaming of /run/ld-so-cache-dir/112bb263f675d03860a13039cd49f7b27bc1750a1838021ad617121117e4ff2c~ to /run/ld-so-cache-dir/112bb263f675d03860a13039cd49f7b27bc1750a1838021ad617121117e4ff2c failed: No such file or directory
error: ldconfig failed, exit status 256
ldconfig: Changing access rights of /run/ld-so-cache-dir/112bb263f675d03860a13039cd49f7b27bc1750a1838021ad617121117e4ff2c~ to 0644 failed: No such file or directory
error: ldconfig failed, exit status 256
ldconfig: Changing access rights of /run/ld-so-cache-dir/112bb263f675d03860a13039cd49f7b27bc1750a1838021ad617121117e4ff2c~ to 0644 failed: No such file or directory
error: ldconfig failed, exit status 256
ldconfig: Renaming of /run/ld-so-cache-dir/112bb263f675d03860a13039cd49f7b27bc1750a1838021ad617121117e4ff2c~ to /run/ld-so-cache-dir/112bb263f675d03860a13039cd49f7b27bc1750a1838021ad617121117e4ff2c failed: No such file or directory
error: ldconfig failed, exit status 256
ldconfig: Renaming of /run/ld-so-cache-dir/112bb263f675d03860a13039cd49f7b27bc1750a1838021ad617121117e4ff2c~ to /run/ld-so-cache-dir/112bb263f675d03860a13039cd49f7b27bc1750a1838021ad617121117e4ff2c failed: No such file or directory
error: ldconfig failed, exit status 256
ldconfig: Changing access rights of /run/ld-so-cache-dir/112bb263f675d03860a13039cd49f7b27bc1750a1838021ad617121117e4ff2c~ to 0644 failed: No such file or directory
error: ldconfig failed, exit status 256
ldconfig: Renaming of /run/ld-so-cache-dir/112bb263f675d03860a13039cd49f7b27bc1750a1838021ad617121117e4ff2c~ to /run/ld-so-cache-dir/112bb263f675d03860a13039cd49f7b27bc1750a1838021ad617121117e4ff2c failed: No such file or directory
error: Can't open generated ld.so.cache
error: ldconfig failed, exit status 256
ldconfig: Changing access rights of /run/ld-so-cache-dir/112bb263f675d03860a13039cd49f7b27bc1750a1838021ad617121117e4ff2c~ to 0644 failed: No such file or directory
error: ldconfig failed, exit status 256
...
```

**Comment by tinywrkb** [comment]
Is this the reason why the Pepper PDF plugin is failing to load in Chrome? If I disable the seccomp sandbox with `--disable-seccomp-filter-sandbox` then it works correctly.  
To reproduce just open a PDF file with the builtin PDF viewer or open the print preview dialog with the destination `Save as PDF`.  
[This is the Flatpak I'm using](https://github.com/tinywrkb/flatpaks/blob/master/com.google.chrome/com.google.chrome.yaml).

**Comment by refi64** [comment]
That Flatpak seems to be using zypak, which isn't really in a suitable state for use with Chrome itself due to a mix of higher memory usage and constantly reproducing this bug. There's a version in progress that will be far better for this, but it's not ready yet, thus I wouldn't recommend trying to use that Chrome Flatpak in the mean time.

**Comment by tinywrkb** [comment]
I just packaged Chrome for testing, not really using it ATM.
Thanks for responding and for working on zypak, I'll keep an eye on its releases.

**Comment by refi64** [comment]
Whoops this is fixed in 1.8.0.

**Comment by alexlarsson** [comment]
Yeah, should work in 1.8
